### PR TITLE
release metadata for `1.12.5` (DMS `B10.0.1`)

### PR DESCRIPTION
**Pull request by zacharyburnett**

### CODE REVIEW

**Comment by zacharyburnett** [comment]
regtest running at https://plwishmaster.stsci.edu:8081/blue/organizations/jenkins/RT%2FJWST-Developers-Pull-Requests/detail/JWST-Developers-Pull-Requests/1005/pipeline

**Review by hbushouse — APPROVED** [review]

File: CHANGES.rst [hunk 1]
```diff
@@ -1,4 +1,4 @@
-1.12.5 (unreleased)
+1.12.5 (2023-10-18)
```
[the remark this answers is not in the document]
Comment (nden) [reply CHANGES.rst:1]: Because I'll make the tag from master and it will go in the release if it's there. I changed the date

**Review by hbushouse — APPROVED** [review]

**Comment by hbushouse** [comment]
sdpdeps CI test is still failing with all those TEST_BIG_DATA undefined errors. What's up with that?

**Comment by braingram** [comment]
It is also failing on `release/1.12.x`
https://github.com/spacetelescope/jwst/actions/runs/6508269333/job/17677309939

**Pull request merged**